spark3/src/main/java/org/apache/iceberg/spark/FileScanTaskSetManager.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import java.util.List;
+import java.util.Map;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.HasTableOperations;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.Pair;
+
+public class FileScanTaskSetManager {
+
+  private static final FileScanTaskSetManager INSTANCE = new FileScanTaskSetManager();
+
+  private final Map<Pair<String, String>, List<FileScanTask>> tasksMap = Maps.newConcurrentMap();
+
+  private FileScanTaskSetManager() {
+  }
+
+  public static FileScanTaskSetManager get() {
+    return INSTANCE;
+  }
+
+  public void stageTasks(Table table, String setID, List<FileScanTask> tasks) {
+    Pair<String, String> id = toID(table, setID);
+    tasksMap.put(id, tasks);
+  }
+
+  public List<FileScanTask> fetchTasks(Table table, String setID) {
+    Pair<String, String> id = toID(table, setID);
+    return tasksMap.get(id);
```

Comment:
I should probably return an empty list here to simplify the validation in the caller.

Reply:
I'm ... not sure. I think if you return empty lists here to represent empty task sets here to represent missing we need to error out on attempting to set an empty set of tasks. Otherwise we won't be able to tell the difference between pushing an empty set and the set never having been set.

I don't think this is a huge deal but I think I would throw an error on trying to staging 0 tasks if we want to return an empty set here ....

I also wouldn't mind throwing explicit exceptions or doing Option types ... monads <3

Reply:
I think I'm going to keep returning null here and will prohibit adding empty sets. Then callers just need to check the tasks are not null.

Reply:
Updated.